openpgp/errors/errors.go

```diff
@@ -9,6 +9,21 @@ import (
 	"strconv"
 )
 
+const (
+	// GenericParsingErrorMessage is a generic error message for parsing errors
+	// to reduce the risk of oracle attacks.
+	GenericParsingErrorMessage = "parsing error"
+)
+
+var (
+	// ErrAEADTagVerification is returned if one of the tag verifications in SEIPDv2 fails
+	ErrAEADTagVerification error = DecryptWithSessionKeyError("AEAD tag verification failed")
+	// ErrMDCHashMismatch is returned if the final tag verification in SEIPDv1 fails
+	ErrMDCHashMismatch error = DecryptWithSessionKeyError("MDC hash mismatch")
+	// ErrMDCMissing is returned if the final tag in SEIPDv1 is missing
+	ErrMDCMissing error = DecryptWithSessionKeyError("MDC packet not found")
```

Comment:
I'd throw a uniform error for `ErrMDCHashMismatch` and `ErrMDCMissing`; depending on when the latter is thrown, it's known to enable an oracle. Not sure if in the current implementation we get to a vulnerable state, but I'd rather not risk it, also thinking of possible future lib changes.

Reply:
Changed in: https://github.com/ProtonMail/go-crypto/pull/237/commits/256571785ad9da96bc4599aa0c70e1d979e309ce